src/main/kotlin/com/manimdsl/ASTExecutor.kt

```diff
@@ -1,55 +1,128 @@
+    private val symbolTable: SymbolTable,
+    private val fileLines: List<String>
+) {
+
+    private val linearRepresentation = mutableListOf<ManimInstr>()
+    private val variableNameGenerator = VariableNameGenerator(symbolTable)
+
+    private val codeBlockVariable: String
+    private val codeTextVariable: String
+    private val pointerVariable: String
+
+    init {
+        pointerVariable = variableNameGenerator.generateNameFromPrefix("pointer")
+        codeBlockVariable = variableNameGenerator.generateNameFromPrefix("code_block")
+        codeTextVariable = variableNameGenerator.generateNameFromPrefix("code_text")
+    }
 
     // Map of all the variables and there current values
     private val variables = mutableMapOf<String, ExecValue>()
 
     // Keep track of next instruction to execute
     private var programCounter = 0
 
-    private fun executeExpression(node: ExpressionNode): ExecValue {
+    private val finalDSLCode = mutableListOf<String>()
+
+    private fun executeExpression(node: ExpressionNode, insideMethodCall :Boolean = false): ExecValue {
         return when (node) {
             is IdentifierNode -> variables[node.identifier]!!
             is NumberNode -> DoubleValue(node.double)
-            is MethodCallNode -> executeMethodCall(node)
-            is ConstructorNode -> executeConstructor(node)
+            is MethodCallNode -> executeMethodCall(node, insideMethodCall)
             is AddExpression -> executeBinaryOp(node) { x, y -> x + y }
             is SubtractExpression -> executeBinaryOp(node) { x, y -> x - y }
             is MultiplyExpression -> executeBinaryOp(node) { x, y -> x * y }
             is PlusExpression -> executeUnaryOp(node) { x -> x }
             is MinusExpression -> executeUnaryOp(node) { x -> -x }
+            else -> EmptyValue
         }
     }
 
-    private fun executeMethodCall(node: MethodCallNode): ExecValue {
+    private fun executeMethodCall(node: MethodCallNode, insideMethodCall :Boolean): ExecValue {
         return when (val ds = variables[node.instanceIdentifier]) {
             is StackValue -> {
-                return when (node.dataStructureMethod.toString()) {
-                    "push" -> DoubleValue(ds.stack.push((executeExpression(node.arguments[0]) as DoubleValue).value))
-                    "pop" -> DoubleValue(ds.stack.pop())
+                return when (node.dataStructureMethod) {
+                    is StackType.PushMethod -> {
+                        val doubleValue = executeExpression(node.arguments[0], true) as DoubleValue
+                        val secondObject = if (ds.stack.empty()) ds.initObject else ds.stack.peek().second
+
+                        // Set animation flags
+                        val animationFlags = AnimationFlags()
+                        animationFlags[OldShape] = doubleValue.manimObject != null
+
+                        val (instructions, newObject) = node.dataStructureMethod.animateMethod(
```

Comment:
This is already done inside a when clause in fact

Reply:
@arj119

Reply:
Working on it